test/python/circuit/test_control_flow.py

```diff
@@ -553,6 +554,72 @@ def test_switch_rejects_input_vars(self):
         with self.assertRaisesRegex(CircuitError, "cannot contain input variables"):
             SwitchCaseOp(target, [(0, good_body), (1, bad_body)])
 
+    def test_box_instantiation(self):
+        """Test that we can instantiate a `BoxOp` directly."""
+        body = QuantumCircuit(3, 1)
+        op = BoxOp(body)
+        self.assertIsInstance(op, ControlFlowOp)
+        self.assertIsInstance(op, Instruction)
+        self.assertEqual(op.name, "box")
+        self.assertEqual(op.num_qubits, body.num_qubits)
+        self.assertEqual(op.num_clbits, body.num_clbits)
+        self.assertEqual(op.params, [body])
+        self.assertEqual(op.blocks, (body,))
+        self.assertEqual(op.duration, None)
+        self.assertEqual(op.label, None)
+
+        op = BoxOp(body, duration=10.0, unit="ms", label="hello, world")
+        self.assertIsInstance(op, ControlFlowOp)
+        self.assertIsInstance(op, Instruction)
+        self.assertEqual(op.name, "box")
+        self.assertEqual(op.num_qubits, body.num_qubits)
+        self.assertEqual(op.num_clbits, body.num_clbits)
+        self.assertEqual(op.params, [body])
+        self.assertEqual(op.blocks, (body,))
+        self.assertEqual(op.duration, 10.0)
+        self.assertEqual(op.unit, "ms")
+        self.assertEqual(op.label, "hello, world")
+
+    def test_box_invalid_params_setter(self):
+        """Verify we catch invalid param settings for BoxOp."""
+        body = QuantumCircuit(3, 1)
+
+        bad_body = QuantumCircuit(2, 1)
+        op = BoxOp(body)
+        with self.assertRaisesRegex(CircuitError, r"different than that of the BoxOp"):
+            op.params = [bad_body]
+        with self.assertRaisesRegex(CircuitError, "body parameter of type QuantumCircuit"):
+            op.params = [None]
```

Comment:
Do you want to have a test of the `setter` working?

Reply:
Done in 1367570070370a4978b5fcd3089a8bb393d4678b.